packages/terra-form-select/src/_Frame.jsx

```diff
@@ -243,12 +248,12 @@ class Frame extends React.Component {
       return;
     }
 
-    if (!this.state.isFocused) {
-      this.setState({ isFocused: true });
+    if (this.props.onFocus && !this.state.isFocused) {
+      this.props.onFocus(event);
     }
 
-    if (this.props.onFocus) {
-      this.props.onFocus(event);
+    if (!this.state.isFocused) {
+      this.setState({ isFocused: true });
```

Comment:
This code changes were made to prevent callback to users onFocus that was firing after onBlur event. 

Since isFocused state variable was meant to check Focus state of the control used it has condition in handleFocus() before doing callback to [users Onfocus()|https://github.com/cerner/terra-core/blob/5864418275108bfa739d907cd1790dad44514def/packages/terra-form-select/src/_Frame.jsx#L252]. 

IsFocus state was set to true on openDropdown() which will be invoked before invocation of handleFocus. Since it was set to true before execution of handleFoucs() condition (!this.state.isFocused) will always fails and prevents even the initial callback to users onFocus() which is supposed to fire when control gets Focus. 

Hence initializing isFocused to false on openDropDown and setting to true in handleFocus after Focus callback has been made to users code. 

Since isFocused is set to true once after initial callback has been it restricts if any attempt made to callback Onfocus again.